app/lib/actions/middleware/watch_delegated_proxy_sub_tasks.rb

```diff
@@ -0,0 +1,65 @@
+module Actions
+  module Middleware
+    class WatchDelegatedProxySubTasks < ::Dynflow::Middleware
+      class CheckOnProxyActions; end
+      POLL_INTERVAL = 10
+
+      def run(event = nil)
+        if event.nil?
+          set_clock
+        elsif event == CheckOnProxyActions
+          check_triggered
+          set_clock
+          action.send(:suspend)
+        end
+        pass event
+      end
+
+      private
+
+      def set_clock
+        action.world.clock.ping action.send(:suspended_action),
+                                POLL_INTERVAL,
+                                CheckOnProxyActions
+      end
+
+      def check_triggered
+        tasks = remote_tasks.triggered.group_by(&:proxy_url)
```

Comment:
I wonder how this works with 10000s of tasks? Shouldn't we use some iteration though batches to spread the tasks over some "smaller' amounts? Maybe it's fine, but from the previous experience, it's good to keep thinking about this option.